moveit_core/planning_interface/include/moveit/planning_interface/planning_response.h

```diff
@@ -37,23 +37,38 @@
 #pragma once
 
 #include <moveit/robot_trajectory/robot_trajectory.h>
+#include <moveit/utils/moveit_error_code.h>
 #include <moveit_msgs/msg/move_it_error_codes.hpp>
 #include <moveit_msgs/msg/motion_plan_response.hpp>
 #include <moveit_msgs/msg/motion_plan_detailed_response.hpp>
 
 namespace planning_interface
 {
+/// \brief Response to a planning query
 struct MotionPlanResponse
 {
-  MotionPlanResponse() : planning_time_(0.0)
+  // Constructor
+  MotionPlanResponse() : trajectory_(nullptr), planning_time_(0.0), error_code_(moveit::core::MoveItErrorCode::FAILURE)
   {
   }
 
+  // Construct a ROS message from struct data
   void getMessage(moveit_msgs::msg::MotionPlanResponse& msg) const;
 
+  // Trajectory generated by the queried planner
   robot_trajectory::RobotTrajectoryPtr trajectory_;
+  // Time to plan the respond to the planning query
   double planning_time_;
-  moveit_msgs::msg::MoveItErrorCodes error_code_;
+  // Result status of the query
+  moveit::core::MoveItErrorCode error_code_;
+  /// The full starting state used for planning
+  moveit_msgs::msg::RobotState start_state_;
+
+  // Enable checking of query success or failure, for example if(response) ...
+  explicit operator bool() const
+  {
+    return bool(error_code_);
```

Comment:
I don't think this is going to work. If `error_code_` is negative, like `int32 PLANNING_FAILED=-1`, it will cast to true.

`int32 SUCCESS=1` also casts to true.

Reply:
```suggestion
    return error_code_ == moveit::core::MoveItErrorCode::SUCCESS;
```

Reply:
This is not the error code value. It's using the `MoveItErrorCode` type which is convertible to bool. See https://github.com/ros-planning/moveit2/blob/main/moveit_core/utils/include/moveit/utils/moveit_error_code.h#L62

Reply:
I suggested to use the same in your PR. https://github.com/ros-planning/moveit2/pull/1605#discussion_r1007826518